File: 054-Rotate Array By K Steps/RotateArrayByKSteps.py
def rotate(nums, k):
    n = len(nums)
    k = k % n  # Handle k > n
    nums[:] = nums[-k:] + nums[:-k]

# Test Input
nums = [1, 2, 3, 4, 5, 6, 7]
k = 3

rotate(nums, k)
print("Rotated Array:", nums)

File: 054-Rotate Array By K Steps/RotateArrayByKSteps.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

void rotate(vector<int>& nums, int k){
    int n = nums.size();
    k = k % n;
    reverse(nums.begin(), nums.end());
    reverse(nums.begin(), nums.begin() + k);
    reverse(nums.begin()+ k, nums.end());

}

int main(){
    vector<int> nums = {1, 2, 3, 4, 5, 6, 7};
    int k = 3;

    rotate(nums, k);
    
    cout<< "Rotated array: ";
    for(int num : nums){
        cout << num << " ";
    }
    return 0;
}